### PR TITLE
Added Support to ID Pools

### PR DESCRIPTION
**Pull request by AsisBagga**
### Description
This PR add support to ID Pools with below endpoints:
GET/rest/id-pools/{poolType}
PUT/rest/id-pools/{poolType}
PUT/rest/id-pools/{poolType}/allocator
GET/rest/id-pools/{poolType}/checkrangeavailability
PUT/rest/id-pools/{poolType}/collector
GET/rest/id-pools/{poolType}/generate
GET/rest/id-pools/{poolType}/validate
PUT/rest/id-pools/{poolType}/validate

### Issues Resolved
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: ov/id_pools.go [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+}
+
+// Gets a pool along with the list of ranges present in it
+func (c *OVClient) GetPoolType(poolType string) (IdPool, error) {
+	var (
+		uri    = "/rest/id-pools/"
+		idPool IdPool
+	)
+
+	if poolType == "" {
```
Comment (VenkateshRavula) [inline ov/id_pools.go:101]: why don't you add this check for previous functions also

File: ov/id_pools.go [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+	log.Debugf("Response update Id Pool %s", data)
+
+	dataResponse := data
+	if err := json.Unmarshal([]byte(data), &t); err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error with task un-marshal: %s", err)
+		return response, err
+	}
+	if err := json.Unmarshal([]byte(dataResponse), &response); err != nil {
```
Comment (VenkateshRavula) [inline ov/id_pools.go:193]: can you please add a comment on what is being validated in this block for my understanding.

File: ov/id_pools.go [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+
+	log.Debugf("Response allocate Ids %s", data)
+
+	if err := json.Unmarshal([]byte(data), &t); err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error with task un-marshal: %s", err)
+		return response, err
+	}
+	if err := json.Unmarshal(data, &response); err != nil {
+		log.Errorf("Error with task un-marshal: %s", err)
```
Comment (VenkateshRavula) [inline ov/id_pools.go:240]: Is this task un-marshal or response un-marshal..?

File: ov/id_pools.go (hunk 2 above)
Comment (AsisBagga) [reply ov/id_pools.go:193]: here we are unmarshalling the json body to array of bytes.

File: ov/id_pools.go (hunk 3 above)
Comment (AsisBagga) [reply ov/id_pools.go:240]: we unmarshall data to &response.

**Review by yuvirani — APPROVED** [review]

File: ov/id_pools.go (hunk 1 above)
Comment (AsisBagga) [reply ov/id_pools.go:109]: done

**Review by VenkateshRavula — APPROVED** [review]

**Pull request merged**